### DISCUSSION TITLE
Is it possible to run a shell script to setup the environment before cmake gets called?

### DISCUSSION THREAD

**Discussion by renyhp** [microsoft/vscode-cmake-tools#2299] [Q&A] [answered]
I have found similar feature requests in the issues, but I'm not sure if this is currently a feature or I should actually open a feature request.

My manual workflow would be

```
source env.sh
cmake ..
```

In other words in the `env.sh` script is setting up the environment for cmake itself.

I have tried using the `environmentSetupScript` in the kit, but I get "bad cmake executable": it looks like `env.sh` is not sourced before the cmake call.

Thanks for any help!

--------------
Oddly enough, even setting `cmake.environment` manually does not work. I have tried `echo`ing the variables that are set by `env.sh` (among which are `PATH`, `LD_LIBRARY_PATH` etc.) and manually prepending them to a call to my cmake executable in a clean shell, and it works properly.
i.e.
```
PATH=... LD_LIBRARY_PATH... /path/to/my/cmake/executable
```

If instead I set the same variables into `cmake.environment` and set `cmake.cmakePath` to the same cmake executable, I still get "Bad CMake executable".

**Comment by bobbrow** [comment, marked as answer]
Sorry for the inconvenience.  This will need to be opened as a feature request.  Currently the extension sources the CMake executable before the `environmentSetupScript` is evaluated.

**Reply by bobbrow** [reply]
I created issue #2301 for this.  Please go and add a 👍 reaction to it to vote for it.
